### PR TITLE
Update hawkBit to Application Version 0.5.0

### PR DESCRIPTION
**Pull request by lharzenetter**
In the current version, the helm chart does not support the latest version of hawkBit that is [0.5.0].

Because of the removal of the vaadin UI, the readyness and liveness probes are broken.
Additionally, by default, the version [0.5.0] uses the [MariaDB Connector][MariaDB-j] instead of the [MySQL Connector][mysql-j].

Thus, I updated the readiness and liveness probes to use the Swagger-UI and the MariaDB connection string by default.

  [0.5.0]: https://github.com/eclipse/hawkbit/releases/tag/0.5.0 
  [MariaDB-j]: https://mariadb.com/kb/en/about-mariadb-connector-j/
  [mysql-j]: https://mvnrepository.com/artifact/com.mysql/mysql-connector-j

### CODE REVIEW

**Comment by calohmn** [comment]
@lharzenetter The CI checks are currently failing here because of a failing `helm test [-n <NS>] eclipse-hawkbit` check.
The issue is that in `charts/hawkbit/templates/tests/test-connection.yaml` there is an HTTP request to the root resource on the hawkBit pod being made, which results in a status 400 response (see https://github.com/eclipse/hawkbit/issues/1832).
Using
```
args:  ['{{ include "hawkbit.fullname" . }}:{{ .Values.service.port }}/swagger-ui/index.html']
```
instead lets the test succeed.
However, as the same endpoint is already used in the liveness/readiness checks, this test wouldn't look very useful. Maybe another endpoint could be used (using the admin credentials).

@strailov @avgustinmm Could you have a look at this PR?

**Comment by avgustinmm** [comment]
> However, as the same endpoint is already used in the liveness/readiness checks, this test wouldn't look very useful. Maybe another endpoint could be used (using the admin credentials).

Using admin credentials you could call:

```shell
curl -X 'GET' \
  'http://localhost:8080/rest/v1/userinfo' \
  -H 'accept: application/json' \
  -H 'Authorization: Basic ...'
```

This shall return info about the tenant and user, e.g.:

```json
{
  "tenant": "DEFAULT",
  "username": "admin"
}
```

This could be a useful liveness/readiness probe since it indicates that the management API is ready to be called up.

**Comment by lharzenetter** [comment]
> This could be a useful liveness/readiness probe since it indicates that the management API is ready to be called up.

I agree.
However, IMHO, we should not use an admin resource as a liveness/readiness probe as you must update it if you use "real" credentials.

**Comment by lharzenetter** [comment]
Do you have any comments on the code? Could you please merge it?

Cheers

**Comment by strailov** [comment]
It looks good for me. Could be merged.

File: charts/hawkbit/templates/tests/test-connection.yaml [hunk 1]
```diff
@@ -11,8 +11,12 @@ metadata:
...
-    - name: wget
-      image: busybox
-      command: ['wget']
-      args:  ['{{ include "hawkbit.fullname" . }}:{{ .Values.service.port }}']
+    - name: curl
+      image: curlimages/curl
+      command: ['curl']
+      args: [
+        "-X", "GET",
+        "-u", "admin:admin",
```
Comment (calohmn) [inline charts/hawkbit/templates/tests/test-connection.yaml:19]: I think instead of `admin` as username/password here, corresponding `Values` references should be put in here (excluding the `{noop}` password prefix).

File: charts/hawkbit/values.yaml [hunk 2]
```diff
@@ -41,7 +41,7 @@ updateStrategy:
   type: Recreate
 
 nameOverride: ""
-fullnameOverride: ""
+fullnameOverride: "eclipse-hawkbit"
```
Comment (calohmn) [inline charts/hawkbit/values.yaml:44]: Is this change needed? I think it's better to keep the old value and let the default be applied (including the release name).

And, could you do a rebase? There was an update of the chart version.

**Review by calohmn — APPROVED** [review]
